### PR TITLE
implemented db connection and BE logic to properly handle api calls

### PR DESCRIPTION
**Pull request by JeffreyChuCPA**
https://www.loom.com/share/e1ad9bdcc61047d98b3504a6273cfc43?sid=87034b38-ddb7-4242-8641-3bfec527e741

### CODE REVIEW

File: client/src/pages/design.tsx [hunk 1]
```diff
@@ -92,6 +92,21 @@ function DesignPage() {
...
+        key_cap_color: themeData.keyCapColor,
+        keyboard_shape: themeData.keyboardShape,
+        keyboard_size: themeData.keyboardSize,
+        keyboard_layout: themeData.keyboardLayout,
+        platform: themeData.platform,
+        owner: auth.user.uid,
+        image_path: imagePath,
+      });
+
+      console.log(keyboardColors);
```
Comment (michaelchu) [inline client/src/pages/design.tsx]: Remove console logs of not needed

File: client/src/util/db.jsx [hunk 2]
```diff
@@ -99,82 +80,56 @@ export function useKeyboardPaginated(page, size = 10) {
...
+
   return response;
 }
 
 /**** HELPERS ****/
 
 // Get response data or throw error if there is one
 function handle(response) {
-  console.log(response);
+  console.log(response.data);
```
Comment (michaelchu) [inline client/src/util/db.jsx]: Here as well

File: server/src/middleware/auth.ts [hunk 3]
```diff
@@ -3,16 +3,9 @@ import admin from 'firebase-admin'
...
-  console.log(idToken);
-  console.log(owner_id);
-  
-  if (!idToken && !owner_id) {
-    return next();
-  }
 
-  if (!idToken && owner_id) {
-    res.status(401).json({ error: 'Unauthorized: Token required for ownerID access'})
+  if (!idToken) {
```
Comment (michaelchu) [inline server/src/middleware/auth.ts:7]: Ideally you should write tests to prove that this works

Comment (JeffreyChuCPA) [reply server/src/middleware/auth.ts:7]: wrote tests in new commit

File: client/src/pages/details.tsx [hunk 4]
```diff
@@ -31,20 +30,17 @@ function DetailsPage() {
...
       <Header />
       <Meta title={`VibeBoard - ${keyboard.theme_name}`} />
-      {isMobile && publicUrl ? (
-        <MobileDetailSection keyboard={keyboard} image={publicUrl} />
+      {isMobile && keyboard.image_path ? (
+        <MobileDetailSection keyboard={keyboard} image={keyboard.image_path} />
       ) : (
         <div className={"container xl:max-w-7xl mx-auto px-4 lg:px-8"}>
-          <DesktopDetailSection keyboard={keyboard} publicUrl={publicUrl} />
+          <DesktopDetailSection keyboard={keyboard} publicUrl={keyboard.image_path} />
```
Comment (alex-sh-wu) [inline client/src/pages/details.tsx:43]: Is `keyboard.image_path` always defined?

If it isn't, this path should check for it.
If it is, then we don't need to check that `keyboard.image_path` is truthy

File: server/src/index.ts [hunk 5]
```diff
@@ -21,6 +21,8 @@ app.get('/', (req: Request, res: Response) => {
...
 })
 
+if (process.env.NODE_ENV !== 'test')  {
+  app.listen(PORT, () => console.log(`Server is live on http://localhost:${PORT}`))
+}
 
-
-app.listen(PORT, () => console.log(`Server is live on http://localhost:${PORT}`))
\ No newline at end of file
+export { app }
```
Comment (alex-sh-wu) [inline server/src/index.ts:28]: Wait, there shouldn't be a file importing the server

File: client/src/pages/details.tsx (hunk 4 above)
Comment (JeffreyChuCPA) [reply client/src/pages/details.tsx:43]: in theory, all keyboard posts received via "useKeyboardByTheme" or any API fetch for keyboards should have an associated image with it. Will remove the truthy check.

**Review by alex-sh-wu — APPROVED** [review]

**Review by michaelchu — APPROVED** [review]
Nice job with the tests, consider refactoring to have a separate file for common mock data in the future.

**Pull request merged**